src/main/java/de/agilecoders/wicket/requirejs/AmdModulesRegistry.java

```diff
@@ -29,91 +29,90 @@
-        }
-
-        return moduleName;
-    }
+	private static final Logger LOG = LoggerFactory.getLogger(RequireJs.class);
+
+	private static final MetaDataKey<ConcurrentMap<String, JavaScriptResourceReference>> MAPPINGS =
+			new MetaDataKey<ConcurrentMap<String, JavaScriptResourceReference>>() {
+	};
+
+	public ConcurrentMap<String, JavaScriptResourceReference> getMappings() {
+		Application application = Application.get();
+		ConcurrentMap<String, JavaScriptResourceReference> mappings = application.getMetaData(MAPPINGS);
+		if (mappings == null) {
+			mappings = new ConcurrentHashMap<>();
+			application.setMetaData(MAPPINGS, mappings);
+		}
+		return mappings;
+	}
+
+	/**
+	 * returns resource reference according to given module name
+	 *
+	 * @param moduleName the module name
+	 * @return resource reference that belongs to module name
+	 */
+	public ResourceReference getReference(String moduleName) {
+		return getMappings().get(name(moduleName));
+	}
+
+	/**
+	 * Registers an AMD module in this registry
+	 *
+	 * @param moduleName The name of the AMD module, e.g. 'my/module'
+	 * @param reference  The JavaScriptResourceReference that loads the content of the module
+	 */
+	public void register(String moduleName, JavaScriptResourceReference reference) {
+		boolean registered = false;
+
+		if (Application.exists()) {
+			ResourceBundles resourceBundles = Application.get().getResourceBundles();
+			HeaderItem bundleItem = resourceBundles.findBundle(JavaScriptHeaderItem.forReference(reference, moduleName));
+
+			if (bundleItem instanceof JavaScriptReferenceHeaderItem) {
+				JavaScriptReferenceHeaderItem jsBundleItem = (JavaScriptReferenceHeaderItem) bundleItem;
+				JavaScriptResourceReference bundleReference = (JavaScriptResourceReference) jsBundleItem.getReference();
+				put(moduleName, bundleReference);
+
+				Map<String, String> mappings = RequireJsConfigHeaderItem.getMappings();
+				mappings.put(moduleName, bundleReference.getName());
+
+				registered = true;
+			}
+		}
+
+		if (!registered) {
+			put(moduleName, reference);
+		}
+	}
+
+	/**
+	 * adds a new module resource reference
+	 *
+	 * @param moduleName the module name
+	 * @param reference the resource reference for module
+	 */
+	private void put(String moduleName, JavaScriptResourceReference reference) {
+		ResourceReference old = getMappings().putIfAbsent(name(moduleName), reference);
+
+		if (old != null && !old.equals(reference)) {
+			LOG.error("'{}' hasn't been registered as AMD module '{}' because there is another module with this name already: {}",
+					reference, moduleName, old);
+		}
+	}
+
+	/**
+	 * assert that given name isn't empty and appends ".js" if missing
+	 *
+	 * @param moduleName the module name to clean up
+	 * @return cleaned module name
+	 */
+	private String name(String moduleName) {
+		Args.notEmpty(moduleName, "moduleName");
+
+		if (!moduleName.endsWith(".js")) {
+			moduleName += ".js";
+		}
+
+		return moduleName;
+	}
```

Comment:
Again big block of re-formatted code.
Sorry but such PR could not be reviewed and thus merged.